src/main/java/io/tiledb/spark/util.java

```diff
@@ -54,6 +57,19 @@ public static Number addEpsilon(Number value, Datatype type) throws TileDBError
     }
   }
 
+  public static URI tryGetArrayURI(TileDBDataSourceOptions tiledbOptions) {
+    Optional<URI> arrayURI;
+    try {
+      arrayURI = tiledbOptions.getArrayURI();
+    } catch (URISyntaxException ex) {
```

Comment:
Why do we need to catch this and throw a different exception?

Reply:
The getArrayURI method in TileDBDataSource options throws this particular exception. This exception is different than the one thrown when the URI is not present.